libsolidity/ast/Types.h

```diff
@@ -208,14 +208,26 @@ class Type
 	virtual bool operator==(Type const& _other) const { return category() == _other.category(); }
 	virtual bool operator!=(Type const& _other) const { return !this->operator ==(_other); }
 
-	/// @returns number of bytes used by this type when encoded for CALL. If it is a dynamic type,
-	/// returns the size of the pointer (usually 32). Returns 0 if the type cannot be encoded
+	/// @returns number of bytes used by this type when encoded for CALL. Cannot be used for
+	/// dynamically encoded types. Returns 0 if the type cannot be encoded
 	/// in calldata.
 	/// If @a _padded then it is assumed that each element is padded to a multiple of 32 bytes.
 	virtual unsigned calldataEncodedSize(bool _padded) const { (void)_padded; return 0; }
+	/// @returns the distance between two elements of this type in an array, tuple or struct.
+	/// For statically-sized types this is the same as calldataEncodedSize(true).
+	/// For dynamically-encoded types this is the distance between two tail pointers, i.e. 32.
+	virtual unsigned calldataHeadIncrement() const { return calldataEncodedSize(); }
+	/// @returns the (minimal) size of the calldata tail for this type. Can only be used for
```

Comment:
Hm - maybe I'll remove "minimal", it might be more confusing than helpful. It just means that for a dynamically sized array it doesn't return the whole size of the array (it can't, since it's not known without actually looking at the dynamic length), but only the size of the length (i.e. 32) which always must be present...

The precise definition in all cases is given below though - so if the tail pointer itself consists of dynamically encoded types, e.g. in ``T[][2][2]``, then in both cases the statically sized array types will have a ``calldataEncodedTailSize`` of 64.

Reply:
Ok. After your explanation, I think "minimal" will make sense if reader knows this function is used to perform some sort of bound check.